Algorithms/Search/KmpSearcher.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using Utilities.Exceptions;
+
+namespace Algorithms.Search
+{
+    /// <summary>
+    /// An implementation of Knuth–Morris–Pratt Algorithm.
+    /// Worst case time complexity: O(n).
```

Comment:
Where n is what?
Also, I suggest writing O(n + k) where n - text length, k - pattern length. It clarifies that auxiliary table construction time is linear with regard to pattern length, which is good.